src/Lib.hs

```diff
@@ -37,41 +33,41 @@ type API = "pgtest" :> Get '[JSON] Foo :<|> "users" :> Get '[JSON] [User] :<|> R
 
 startApp :: RIO Env ()
 startApp = do
-  env <- ask
-  let localPort = view portL env
-  logInfo $ "Stage: " <> (displayShow $ stage env)
+  localPort <- view portL
+  theStage <- view stageL
+  logInfo $ "Stage: " <> displayShow theStage
   logInfo $ "port" <> displayShow localPort
-  application <- app
+  application <- buildApp
   liftIO $ run localPort application
 
-app :: WithStage env => WithConnectionPool env => RIO env Application
-app = do
+buildApp :: (WithStage env, WithConnectionPool env) => RIO env Application
+buildApp = do
+  localstage <- view stageL
   env <- ask
-  let localstage = view stageL env
-  localserver <- server
-  return $ sslRedirect localstage . corsMiddleware $ serve api localserver
-
+  return . sslRedirect localstage . corsMiddleware . serve api . hoist $ env
 
 api :: Proxy API
 api = Proxy
-server :: WithConnectionPool env => (RIO env (Server API))
-server = do
-  pg <- pgtest
-  urs <- users
-  return  $ (return pg :<|> return urs :<|> serveDirectoryWith (staticSettings "react"))
+
+server :: WithConnectionPool env => ServerT API (RIO env)
+server = pgtest :<|> users :<|> serveDirectoryWith (staticSettings "react")
 
 users :: RIO env [User]
 users = return [ User 1 "Isaac" "Newton"
-        , User 2 "Albert" "Einstein"
-        , User 3 "Stephen" "Hawking"
-        ]
+               , User 2 "Albert" "Einstein"
+               , User 3 "Stephen" "Hawking"
+               ]
 
 pgtest :: WithConnectionPool env => RIO env Foo
 pgtest = do
-  env <- ask
-  let conns = view connectionPoolL env
+  conns <- view connectionPoolL
   liftIO (executePg conns)
   where
     executePg conns = withResource conns $ \conn -> do
       [Only i] <- query_ conn "select 2 + 2"
       return (Foo i)
+
+hoist :: forall env. WithConnectionPool env => env -> Server API
+hoist env = hoistServer api nat server
```

Comment:
`hoistServer` takes a `natural transformation` (the `forall x. m x -> n x) and this natural transformation defines how to "run" your custom monad (`RIO env`) and converts it into a `Handler` from servant.  That's the magic that makes it work